### PR TITLE
Add "Work items" documentation guidance to Documentation section

### LINKED ISSUE #603: Add "Work items" documentation guidance to Documentation section
After the new Documentation section is created (#533), add a new "Work items" document under "What documentation should exist" (in documentation/guidance folder).

Ideas for content:

- Epics/Features/User Stories/Tasks/Bugs.
  - Title.
  - Description.
  - Acceptance Criteria.
  - Estimation.
  - Discussion.
- Stories/tasks include:
  - Design reviews.
  - Observability design components.
- Actions from Retrospectives.

### PR DESCRIPTION
**Pull request by magencio**
## What are you trying to address

Add a new "Work items" document under "What documentation should exist" (in documentation/guidance folder).

Closes #603 

## For all pull requests

- [X] Link to the issue you are solving (so it gets closed)
- [X] Label the pull request with the appropriate area(s)
- [X] Assign potential reviewers (you may also want to contact them on Teams to ensure timely reviews)
- [X] Assign the project - Engineering Playbook Backlog
- [X] Assign the pull request to yourself

### CODE REVIEW

File: docs/documentation/guidance/work-items.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
+# Work Items
+
+While many teams can work with a flat list of items, sometimes it helps to group related items into a hierarchical structure. You can use portfolio backlogs to bring more order to your backlog.
+
+**Agile** process backlog work item hierarchy:
+
+![agile-artifacts](./images/scrum-artifacts.png)
```
Comment (tompaana) [inline docs/documentation/guidance/work-items.md:7]:
```suggestion
![agile-artifacts](./images/agile-artifacts.png)
```

File: docs/documentation/guidance/work-items.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+## Related information
+
+- [Best practices for Agile project management - Azure Boards | Microsoft Docs](https://docs.microsoft.com/en-us/azure/devops/boards/best-practices-agile-project-management?view=azure-devops&tabs=agile-process).
+- [Define features and epics, organize backlog items - Azure Boards | Microsoft Docs](https://docs.microsoft.com/en-us/azure/devops/boards/backlogs/define-features-epics?view=azure-devops&tabs=scrum-process).
+- [Create your product backlog - Azure Boards | Microsoft Docs](https://docs.microsoft.com/en-us/azure/devops/boards/backlogs/create-your-backlog?view=azure-devops&tabs=agile-process).
+- [Add tasks to support sprint planning - Azure Boards | Microsoft Docs](https://docs.microsoft.com/en-us/azure/devops/boards/sprints/add-tasks?view=azure-devops).
+- [Define, capture, triage, and manage bugs or code defects - Azure Boards | Microsoft Docs](https://docs.microsoft.com/en-us/azure/devops/boards/backlogs/manage-bugs?view=azure-devops).
+- [Add and manage issues or impediments - Azure Boards | Microsoft Docs](https://docs.microsoft.com/en-us/azure/devops/boards/backlogs/manage-issues-impediments?view=azure-devops).
```
Comment (tompaana) [inline docs/documentation/guidance/work-items.md:105]:
```suggestion
- [Best practices for Agile project management - Azure Boards | Microsoft Docs](https://docs.microsoft.com/azure/devops/boards/best-practices-agile-project-management?view=azure-devops&tabs=agile-process).
- [Define features and epics, organize backlog items - Azure Boards | Microsoft Docs](https://docs.microsoft.com/azure/devops/boards/backlogs/define-features-epics?view=azure-devops&tabs=scrum-process).
- [Create your product backlog - Azure Boards | Microsoft Docs](https://docs.microsoft.com/azure/devops/boards/backlogs/create-your-backlog?view=azure-devops&tabs=agile-process).
- [Add tasks to support sprint planning - Azure Boards | Microsoft Docs](https://docs.microsoft.com/azure/devops/boards/sprints/add-tasks?view=azure-devops).
- [Define, capture, triage, and manage bugs or code defects - Azure Boards | Microsoft Docs](https://docs.microsoft.com/azure/devops/boards/backlogs/manage-bugs?view=azure-devops).
- [Add and manage issues or impediments - Azure Boards | Microsoft Docs](https://docs.microsoft.com/azure/devops/boards/backlogs/manage-issues-impediments?view=azure-devops).
```

**Review by tompaana — APPROVED** [review]
Clear and concise! I added few suggestions to fix minor issues.

**Review by TessFerrandez — APPROVED** [review]
LGTM!

**Pull request merged**